proposals/partial-events-and-constructors.md

```diff
@@ -0,0 +1,293 @@
+                format.GetNonNullHandle(nameof(format)),
+                packetCapacity),
+            "initWithFormat:packetCapacity:");
+    }
+
+    public partial event EventHandler Created
+    {
+        add { /* ... */ }
+        remove { /* ... */ }
+    }
+}
+```
+
+## Detailed design
+
+<!-- This is the bulk of the proposal. Explain the design in enough detail for somebody familiar with the language to understand, and for somebody familiar with the compiler to implement, and include examples of how the feature is used. This section can start out light before the prototyping phase but should get into specifics and corner-cases as the feature is iteratively designed and implemented. -->
+
+### General
+
+Event declaration syntax ([§15.8.1][event-syntax]) is extended to allow the `partial` modifier:
+
+```diff
+ event_declaration
+     : attributes? event_modifier* 'event' type variable_declarators ';'
+     | attributes? event_modifier* 'event' type member_name
+         '{' event_accessor_declarations '}'
+     ;
+
+ event_modifier
+     : 'new'
+     | 'public'
+     | 'protected'
+     | 'internal'
+     | 'private'
+     | 'static'
+     | 'virtual'
+     | 'sealed'
+     | 'override'
+     | 'abstract'
+     | 'extern'
++    | 'partial'
+     | unsafe_modifier
+     ;
+```
+
+Instance constructor declaration syntax ([§15.11.1][ctor-syntax]) is extended to allow the `partial` modifier:
+
+```diff
+ constructor_declaration
+     : attributes? constructor_modifier* constructor_declarator constructor_body
+     ;
+ 
+ constructor_modifier
+     : 'public'
+     | 'protected'
+     | 'internal'
+     | 'private'
+     | 'extern'
++    | 'partial'
+     | unsafe_modifier
+     ;
+```
+
+Note that the `partial` modifier can be specified anywhere among the modifiers unlike today in methods and types where it must be the last one.
+There is [a proposal][partial-ordering] to relax ordering of modifiers w.r.t. `partial` in method and type declarations as well.
+
+An event declaration with the `partial` modifier is said to be a *partial event declaration*
+and it is associated with one or more *partial events* with the specified name
+(note that one event declaration without accessors can define multiple events).
+
+A constructor declaration with the `partial` modifier is said to be a *partial constructor declaration*
+and it is associated with a *partial constructor* with the specified signature.
+
+A partial event declaration is said to be an *implementing declaration*
+when it specifies the `event_accessor_declarations` or it has the `extern` modifier.
+Otherwise, it is a *defining declaration*.
+
+A partial constructor declaration is said to be a *defining declaration*
+when it has the semicolon body and it lacks the `extern` modifier.
+Otherwise, it is an *implementing declaration*.
+
+```cs
+partial class C
+{
+    // defining declarations
+    partial C();
+    partial C(int x);
+    partial event Action E, F;
```

Comment:
It looks like when a multi-declarator event declaration has attributes, *all* the event symbols get those same attributes. That's interesting.  [SharpLab](https://sharplab.io/#v2:EYLgtghglgdgPgAQEwEYCwAoBBmABM3AYVwG9NNdLcBtAQQBd6AnAXQqp1wFMA3LmevhRJcAURQAaMUgDc5DFXx4EAFlwBZABQBKUu0oBfTEYyZY9LkxgQANvhENmjplGABXC7hC5nrj10wyDAMgA===).

So, it seems really rare for people to actually do this, but, I can at least imagine such a partial declaration having a functional purpose, and am not really seeing a reason to block it.